tests/ui/recursion/recursive-enum-box.rs

```diff
@@ -0,0 +1,12 @@
+//@ run-pass
+// A smoke test for recursive enum structures using Box<T>.
+// This test constructs a linked list-like structure to verify memory allocation and ownership.
+// Originally introduced in 2010, this is one of Rust’s earliest test cases.
+
+#![allow(non_camel_case_types)]
+
+enum list { #[allow(dead_code)] cons(isize, Box<list>), nil, }
+
+pub fn main() {
+    list::cons(10, Box::new(list::cons(11, Box::new(list::cons(12, Box::new(list::nil))))));
+}
```

Comment:
Suggestion: use canonical naming for this test, `non_camel_case_types` here is unnecessarily distracting. Please also run `rustfmt` against this test. i.e.

```rs
enum List {
    Cons(isize, Box<List>),
    Nil,
}

pub fn main() {
    List::Cons(
        10,
        Box::new(List::Cons(
            11,
            Box::new(List::Cons(12, Box::new(List::Nil))),
        )),
    );
}
```

Reply:
Would it be a good idea to modify the test code so that we can remove the #[allow(dead_code)] attribute as well?

Reply:
Yes, I wouldn't bother trying to get rid of the `dead_code`, it's the non camel case types and variants that are needlessly distracting